tests/miral/test_window_manager_tools.cpp

```diff
@@ -0,0 +1,290 @@
+    void set_depth_layer(MirDepthLayer depth_layer) override { depth_layer_ = depth_layer; }
+
+    std::string name_;
+    MirWindowType type_;
+    mir::geometry::Point top_left_;
+    mir::geometry::Size size_;
+    MirWindowState state_ = mir_window_state_restored;
+    MirDepthLayer depth_layer_;
+};
+
+struct StubStubSession : mir::test::doubles::StubSession
+{
+    mir::frontend::SurfaceId create_surface(
+        mir::scene::SurfaceCreationParameters const& params,
+        std::shared_ptr<mir::frontend::EventSink> const& /*sink*/) override
+    {
+        auto id = mir::frontend::SurfaceId{next_surface_id.fetch_add(1)};
+        auto surface = std::make_shared<StubSurface>(
+            params.name,
+            params.type.value(),
+            params.top_left,
+            params.size,
+            params.depth_layer.is_set() ?
+                params.depth_layer.value()
+                : mir_depth_layer_application);
+        surfaces[id] = surface;
+        return id;
+    }
+
+    std::shared_ptr<mir::scene::Surface> surface(mir::frontend::SurfaceId surface) const override
+    {
+        return surfaces.at(surface);
+    }
+
+private:
+    std::atomic<int> next_surface_id;
+    std::map<mir::frontend::SurfaceId, std::shared_ptr<mir::scene::Surface>> surfaces;
+};
+
+struct MockWindowManagerPolicy : miral::CanonicalWindowManagerPolicy
+{
+    using miral::CanonicalWindowManagerPolicy::CanonicalWindowManagerPolicy;
+
+    bool handle_touch_event(MirTouchEvent const* /*event*/) { return false; }
+    bool handle_pointer_event(MirPointerEvent const* /*event*/) { return false; }
+    bool handle_keyboard_event(MirKeyboardEvent const* /*event*/) { return false; }
+
+    MOCK_METHOD1(advise_new_window, void (miral::WindowInfo const& window_info));
+    MOCK_METHOD2(advise_move_to, void(miral::WindowInfo const& window_info, mir::geometry::Point top_left));
+    MOCK_METHOD2(advise_resize, void(miral::WindowInfo const& window_info, mir::geometry::Size const& new_size));
+    MOCK_METHOD1(advise_raise, void(std::vector<miral::Window> const&));
+    MOCK_METHOD1(advise_output_create, void(miral::Output const&));
+    MOCK_METHOD2(advise_output_update, void(miral::Output const&, miral::Output const&));
+    MOCK_METHOD1(advise_output_delete, void(miral::Output const&));
+
+    void handle_request_drag_and_drop(miral::WindowInfo& /*window_info*/) {}
+    void handle_request_move(miral::WindowInfo& /*window_info*/, MirInputEvent const* /*input_event*/) {}
+    void handle_request_resize(miral::WindowInfo& /*window_info*/, MirInputEvent const* /*input_event*/, MirResizeEdge /*edge*/) {}
+    mir::geometry::Rectangle confirm_placement_on_display(const miral::WindowInfo&, MirWindowState, mir::geometry::Rectangle const& new_placement)
+    {
+        return new_placement;
+    }
+};
+
+struct TestDisplayConfigurationObserver : mir::ObserverRegistrar<mir::graphics::DisplayConfigurationObserver>
```

Comment:
I'd call TestDisplayConfigurationObserver FakeDisplayConfigurationObserverRegistrar as that better describes its use here.

Reply:
Done